src/main/java/hudson/remoting/Launcher.java

```diff
@@ -256,6 +310,14 @@ public void run() throws Exception {
             if (this.noKeepAlive) {
                 jnlpArgs.add("-noKeepAlive");
             }
+            if (this.workDir != null) {
+                jnlpArgs.add("-workDir");
+                jnlpArgs.add(workDir.getPath());
+                jnlpArgs.add("-internalDir");
+                jnlpArgs.add(internalDir);
+                jnlpArgs.add("-failIfWorkDirIsMissing");
+                jnlpArgs.add(Boolean.toString(failIfWorkDirIsMissing));
+            }
             if (candidateCertificates != null && !candidateCertificates.isEmpty()) {
```

Comment:
There is https://issues.jenkins-ci.org/browse/JENKINS-39817 for it. I'm not a big fan of adding this option since we are going to have the `workDir` now. But maybe it's good for the code consistency